src/Avalonia.Input/Pointer.cs

```diff
@@ -9,23 +11,40 @@ public class Pointer : IPointer, IDisposable
         private static int s_NextFreePointerId = 1000;
         public static int GetNextFreeId() => s_NextFreePointerId++;
         
-        public Pointer(int id, PointerType type, bool isPrimary, IInputElement implicitlyCaptured)
+        public Pointer(int id, PointerType type, bool isPrimary)
         {
             Id = id;
             Type = type;
             IsPrimary = isPrimary;
-            ImplicitlyCaptured = implicitlyCaptured;
-            if (ImplicitlyCaptured != null)
-                ImplicitlyCaptured.DetachedFromVisualTree += OnImplicitCaptureDetached;
         }
 
         public int Id { get; }
 
+        IInputElement FindCommonParent(IInputElement control1, IInputElement control2)
```

Comment:
There is already `VisualExtensions.FindCommonVisualAncestor`: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Visuals/VisualTree/VisualExtensions.cs#L46 - does this not work properly?

Reply:
It lacks `OfType<IInputElement>()`, since we aren't interested in just visuals.